test/suites/integration/ami_test.go

```diff
@@ -299,6 +302,18 @@ var _ = Describe("AMI", func() {
 			Expect(err).ToNot(HaveOccurred())
 			Expect(string(actualUserData)).To(ContainSubstring("kube-api-qps = 30"))
 		})
+		It("should have the AWSNodeTemplateStatus for AMIs", func() {
+			provider := awstest.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{
+				AWS: v1alpha1.AWS{
+					SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+					SubnetSelector:        map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+				},
+				AMISelector: map[string]string{"aws-ids": customAMI},
+			})
+
+			env.ExpectCreated(provider)
+			EventuallyExpectAMIs(provider, customAMI)
```

Comment:
I think we need to have a higher bar of enforcement here. i.e. we should enforce the entirety of the status. That way we ensure the entire functionality of the feature

Reply:
During the integration test, we actually don't know what the name will be as we get the from EC2. I will add a check for requirements.